### DISCUSSION TITLE
Database of phreeqc in GeoChemFoam

### DISCUSSION THREAD

**Discussion by Mohamadreza1998** [GeoChemFoam/GeoChemFoam#23] [Q&A] [answered]
I am looking for the way I can change the database of phreeqc?

I want to use PITZER database instead of phreeqc database. To do this, I copied all data of PITZER database in Geochem.dat of GeoChemFoam. The simulation process fails when utilizing this database.

I will be happy you give me your opinion (If any) on this case.

Thanks.

**Comment by julienmaes** [comment]
It could be that some of the phreeqc keyword in the pitzer database are not supported by GCFoam

**Comment by Mohamadreza1998** [comment]
Thanks.

Could be add a keyword in GCFoam?

This database in GCFoam use for pressure dependence of reactions but the new database of phreeqc use for temperature and pressure dependence of reactions, and the specific conductance and viscosity of the solution.

For example in the new database the viscosity was added in database:
CO3-2 + H+ = HCO3-
	-log_k	10.329
	-delta_h -3.561	kcal
	-analytic  107.8871  0.03252849  -5151.79  -38.92561  563713.9
	-gamma	5.4      0
	-dw	1.18e-9  -182  0.351  -4.94
	-Vm  9.03  -7.03e-2  -13.38  0  2.05  0  0  128  0  0.8242
	-dw	1.18e-9  -182  0.351  -4.94
	-viscosity  0  0.117  -2.91e-2  0  0  0  0.896

I copied all data of new database of phreeqc in Geochem.dat file but the simulation process fails because as you said, the GCFoam does not supported some keywords.

So, Is it possible to add viscosity to GCFoam?

**Comment by GeoChemFoam** [comment]
Hi Mohamed,

The code is open source so you are more than welcome to add it yourself to your version.

Hannah

On Oct 9, 2023, at 14:36, Mohamadreza1998 ***@***.***> wrote:

﻿
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Thanks.

Could be add a keyword in GCFoam?

This database in GCFoam use for pressure dependence of reactions but the new database of phreeqc use for temperature and pressure dependence of reactions, and the specific conductance and viscosity of the solution.

For example in the new database the viscosity was added in database:
CO3-2 + H+ = HCO3-
-log_k 10.329
-delta_h -3.561 kcal
-analytic 107.8871 0.03252849 -5151.79 -38.92561 563713.9
-gamma 5.4 0
-dw 1.18e-9 -182 0.351 -4.94
-Vm 9.03 -7.03e-2 -13.38 0 2.05 0 0 128 0 0.8242
-dw 1.18e-9 -182 0.351 -4.94
-viscosity 0 0.117 -2.91e-2 0 0 0 0.896

I copied all data of new database of phreeqc in Geochem.dat file but the simulation process fails because as you said, the GCFoam does not supported some keywords.

So, Is it possible to add viscosity to GCFoam?

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/23#discussioncomment-7230055>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFWZC6OAIPKUGXFGXO3X6PVUNAVCNFSM6AAAAAA5XEPAZKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TEMZQGA2TK>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Reply by Mohamadreza1998** [reply, marked as answer]
Thank you very much.
